src/lib/parser/YamlParser.ts

```diff
@@ -0,0 +1,27 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { HostParameterEntry } from "@microsoft/powerplatform-cli-wrapper/dist/host/IHostAbstractions";
+import fs = require('fs');
+import yaml = require('js-yaml');
+import path = require("path");
+
+export class YamlParser {
+    public getHostParameterEntries(workingDir: string, actionFolder: string): Record<string, HostParameterEntry> {
+        const parameterMap: Record<string, HostParameterEntry> = {};
+        try {
+            const file = path.resolve(workingDir, `./${actionFolder}/action.yml`);
+            const fileContents = fs.readFileSync(file, 'utf8');
+            const data = yaml.load(fileContents) as any;
+            for (const [name, inputInfo] of Object.entries(data.inputs)) {
+                const typedData = {
+                    name: name,
+                    required: (inputInfo as any).required ?? false,
```

Comment:
yep, as discussed for a future PR:  adding interface definitions for the 2 shapes within the yaml will improve both the readability of the code and the robustness